src/app_work.c

```diff
@@ -51,22 +51,25 @@ void app_work_sensor_read(void)
 	/* For this demo, we just send Hello to Golioth */
 	static uint8_t counter;
 
-	LOG_INF("Sending hello! %d", counter);
-
-	err = golioth_send_hello(client);
-	if (err) {
-		LOG_WRN("Failed to send hello!");
+	if (golioth_is_connected(client)) {
```

Comment:
This is a substantive change and should be in its own commit. However, I wonder if we can remove this from the current PR and have some more thought about the change itself. Here's why I say that:

* I currently like the behavior that an error is thrown if it's time to write to the cloud and there is no connection available to Golioth
  * This change currently doesn't inform the user that a write to the cloud was missed. If we were to use this change, an error should still be printed that the cloud write was not possible
* If we just want to prevent trying to write to the cloud because there is no connection available, we could use a semaphore to block until the connection is established (I do not like this behavior for reference designs)
* I'm unsure how the `golioth_is_connected()` is implemented, but I have some memory of times in the past where disconnect events didn't reset this value. That may not be true now (or I may be remembering wrong) but this should be investigated.